File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
      vector<int> ans;
        int n=nums.size();
        int pos=0;
        int neg=0;
       while(pos<n||neg<n)
        {   while(pos<n&&nums[pos]<0)
        {
            pos++;
        }
         if(pos!=n) 
         { 
             ans.push_back(nums[pos]);
             pos++;
         }
            
            
                while(neg<n&&nums[neg]>0)
        {
            neg++;
        }
         if(neg!=n) 
         { 
             ans.push_back(nums[neg]);
             neg++;
         }
            
        }
        return ans;
    }
};

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int arr[]=new int[nums.length];
        int pos=0,neg=1;
        for(int i=0;i<nums.length;i++)
        {
            if(nums[i]<0)
            {
                arr[neg]=nums[i];
                neg+=2;
            }
            else{
                arr[pos]=nums[i];
                pos+=2;
            }
        }
        return arr;
        
    }
}